src/aspire/basis/pswf_2d.py

```diff
@@ -1,5 +1,320 @@
+        images[self.get_points_inside_the_circle(), :] = flatten_images
+        images = np.transpose(images, axes=(1, 0, 2))
+        return np.real(images)
+
+    def get_samples_as_images(self):
+        print("get_samples_as_images is not supported.")
+        return -1
+
+    def get_angular_frequency(self):
+        return self.angular_frequency
+
+    def get_num_prolates(self):
+        # TODO: fix once Itay gives the samples in the correct way
+        print("get_samples_as_images is not supported.")
+        return -1
+
+    def get_non_neg_freq_inds(self):
+        return self.non_neg_freq_inds
+
+    def get_zero_freq_inds(self):
+        return self.zero_freq_inds
+
+    def get_pos_freq_inds(self):
+        return self.pos_freq_inds
+
+    def get_neg_freq_inds(self):
+        ValueError('no negative frequencies')
+
+    def init_pswf_func2d(self, c, eps):
+        """
+        initialize the whole set of PSWF functions with the input bandlimit and error.   
+        :param c: bandlimit  
+        :param eps: error tolerance
+        :return:
+            alpha_all (list of arrays):
+            alpha = alpha_all[i] contains all the eigenvalues for N=i such that lambda > eps,
+                    where lambda is the normalized  alpha values (i.e. lambda is between 0 and 1) ,
+                    given by lambda=sqrt(c*np.absolute(alpha)/(2*pi)).
+            d_vec_all (list of 2D lists): the corresponding eigenvectors for alpha_all
+            n_order_length_vec (list of ints): n_order_length_vec[i] = len(alpha_all[i])
+        """
+        d_vec_all = []
+        alpha_all = []
+        n_order_length_vec = []
+
+        m = 0
+        n = int(np.ceil(2 * c / np.pi))
+        #n = self.resolution + 1
+        x, w = leggauss_0_1(n)
+        #x, w = leggauss_0_1(1000)
+
+        cons = c / 2 / np.pi
+        while True:
+            alpha, d_vec, a = self.pswf_func2d(m, n, c, eps, x, w)
+
+            # should check this lambda
+            lambda_var = np.sqrt(cons * np.absolute(alpha))
+
+            n_end = np.where(lambda_var <= eps)[0]
+
+            if len(n_end) != 0:
+                n_end = n_end[0]
+                if n_end == 0:
+                    break
+                n_order_length_vec.extend([n_end])
+                alpha_all.append(alpha[:n_end])
+                d_vec_all.append(d_vec[:, :n_end])
+                m += 1
+                # print("generating pswfs for angular index: {}".format(m))
+                n = n_end + 1
+            else:
+                n *= 2
+                x, w = leggauss_0_1(n)
+
+        return d_vec_all, alpha_all, n_order_length_vec
+
+    def evaluate_pswf2d_all(self, x, y, max_ns):
+        """
+        Evaluate the numerical values of PSWF basis functions for all N's, up to certain given n for each N
+        :param x: Radial part to evaluate
+        :param y: Phase part to evaluate
```

Comment:
Aren't better names `r` and `theta`? Why `x` and `y`?

Reply:
renamed.